plugins/Xpressive/Xpressive.cpp

```diff
@@ -572,20 +572,21 @@ void XpressiveView::expressionChanged() {
 			expr.add_cyclic_vector("W2",e->graphW2().samples(),e->graphW2().length());
 			expr.add_cyclic_vector("W3",e->graphW3().samples(),e->graphW3().length());
 		}
-		expr.setIntegrate(&i,sample_rate);
+        expr.setIntegrate(&frame_counter,sample_rate);
 		expr.add_constant("srate",sample_rate);
 
 		const bool parse_ok=expr.compile();
 
 		if (parse_ok) {
 			e->exprValid().setValue(0);
-			const auto length = static_cast<std::size_t>(m_raw_graph->length());
+            const unsigned int length = static_cast<unsigned int>(m_raw_graph->length());
```

Comment:
You should maybe revert this one line of change. For size_t, the std:: prefix is unnecessary tho.

Reply:
Why revert?

Reply:
Auto is fine to use here. Also, like i said in the other line, size_t and unsigned int is identical.

Reply:
Correct me if I am wrong, they are identical on 32 bit systems but not on 64 bit. I want length to be declared with the same type of frame_counter, so I will not get warnings when comparing the two

Reply:
What's the point if m_raw_graph->length() returns int?

Reply:
@sakertooth you are welcome to change the type of the variable m_length in the file Graph.h . 
there is no benefit to use size_t when the value originated as int. 
Changing m_length variable into size_t sounds like a long adventure for me.
For this reason, I shall, for now, stick with unsigned int. I just hope Graph.h length never returns a negative value.